cloud-functions/cf-file-scanner/src/main/java/org/broadinstitute/ddp/clamav/Client.java

```diff
@@ -0,0 +1,271 @@
+package org.broadinstitute.ddp.clamav;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+public class Client {
+    private static final Logger logger = Logger.getLogger(Client.class.getName());
+
+    private class Defaults {
+        static final int PORT = 3310;
+        static final int SO_TIMEOUT = 30000; // 30 seconds
+        static final int MAX_CHUNK_SIZE = 1000000; // 1MB
+    }
+
+    private enum Delimiter {
+        NULL,
+        NEWLINE;
+
+        public byte[] modeFlag() {
+            switch (this) {
+                case NULL:
+                    return new byte[] { 'z' };
+                case NEWLINE:
+                    return new byte[] { 'n' };
+                default:
+                    throw new RuntimeException("unreachable");
+            }
+        }
+
+        public byte[] lineTerminator() {
+            switch (this) {
+                case NULL:
+                    return new byte[] { 0x00 };
+                case NEWLINE:
+                    return new byte[] { 0x0A };
+                default:
+                    throw new RuntimeException("unreachable");
+            }
+        }
+    }
```

Comment:
Pretty sure this is a Swift-ism working it's way in. I can propose an alternative using `getModeFlag()` and `getLineTerminator()`, and using a ctor to pass in the values, but is this idiomatic enum use for Java?